platform/pulpcore/app/upload.py

```diff
@@ -0,0 +1,33 @@
+import hashlib
+from django.core.files.uploadhandler import TemporaryFileUploadHandler
+from django.core.files.uploadedfile import TemporaryUploadedFile
+
+
+class PulpTemporaryUploadedFile(TemporaryUploadedFile):
+    """
+    A file uploaded to a temporary location in Pulp.
+    """
+
+    def __init__(self, name, content_type, size, charset, content_type_extra=None):
+        self.hashers = {}
+        for hasher in hashlib.algorithms_guaranteed:
+            self.hashers[hasher] = getattr(hashlib, hasher)()
+        super(PulpTemporaryUploadedFile, self).__init__(name, content_type, size, charset, content_type_extra)
+
+
+class PulpFileUploadHandler(TemporaryFileUploadHandler):
+    """
+    Upload handler that streams data into a temporary file.
+    """
+    def new_file(self, file_name, *args, **kwargs):
+        """
+        Create the file object to append to as data is coming in.
+        """
+        super(PulpFileUploadHandler, self).new_file(file_name, *args, **kwargs)
```

Comment:
In an act of mercy, Python 3 no longer requires us to pass stuff into ``super()``. :smile: